packages/wonder-blocks-data/util/request-tracking.test.js

```diff
@@ -63,89 +70,254 @@ describe("./request-tracking.js", () => {
 
                 // Act
                 requestTracker.trackDataRequest(fakeHandler, options);
-                const {
-                    trackedHandlers,
-                } = requestTracker.tempGetTrackedRequestsAndHandlers();
+                await requestTracker.fulfillTrackedRequests();
 
                 // Assert
-                expect(trackedHandlers).toHaveProperty("MY_TYPE", fakeHandler);
+                expect(fakeHandler.fulfillRequest).toHaveBeenCalledWith(
+                    options,
+                );
+                expect(fakeHandler.fulfillRequest).toHaveBeenCalledTimes(1);
             });
 
-            it("should add a request to the cache if it is not present", () => {
+            it("should track each matching request once", async () => {
                 // Arrange
-                const requestTracker = new RequestTracker();
+                const requestTracker = createRequestTracker();
                 const fakeHandler: IRequestHandler<any, any> = {
-                    fulfillRequest: jest.fn(),
+                    fulfillRequest: jest.fn(() => Promise.resolve(null)),
                     getKey: jest.fn().mockReturnValue("MY_KEY"),
                     invalidateCache: () => false,
                     type: "MY_TYPE",
                 };
-                const options = {these: "are options"};
+                const options1 = {these: "are options"};
+                const options2 = {these: "are options"};
 
                 // Act
-                requestTracker.trackDataRequest(fakeHandler, options);
-                const {
-                    trackedRequests,
-                } = requestTracker.tempGetTrackedRequestsAndHandlers();
+                requestTracker.trackDataRequest(fakeHandler, options1);
+                requestTracker.trackDataRequest(fakeHandler, options2);
+                await requestTracker.fulfillTrackedRequests();
 
                 // Assert
-                expect(trackedRequests).toHaveProperty("MY_TYPE", {
-                    MY_KEY: options,
-                });
+                expect(fakeHandler.fulfillRequest).toHaveBeenCalledWith(
+                    options1,
+                );
+                expect(fakeHandler.fulfillRequest).toHaveBeenCalledTimes(1);
```

Comment:
This is because `getKey` always returns the same value?

Reply:
No, it's because `getKey` should always return the same value when given the same options. In this test, I've hard coded `getKey` to always return the same value, but in reality, we'd be basing that key off the options passed.

Reply:
I'm updating the test to make that clearer.